opentelemetry-exporter-geneva/opentelemetry-exporter-geneva/src/logs/exporter.rs

```diff
@@ -1,25 +1,33 @@
 use core::fmt;
+use futures::stream::{self, StreamExt};
 use geneva_uploader::client::GenevaClient;
 use opentelemetry_proto::transform::common::tonic::ResourceAttributesWithSchema;
 use opentelemetry_proto::transform::logs::tonic::group_logs_by_resource_and_scope;
 use opentelemetry_sdk::error::{OTelSdkError, OTelSdkResult};
 use opentelemetry_sdk::logs::LogBatch;
-use std::sync::atomic;
+use std::sync::{atomic, Arc};
 
 /// An OpenTelemetry exporter that writes logs to Geneva exporter
 pub struct GenevaExporter {
     resource: ResourceAttributesWithSchema,
     _is_shutdown: atomic::AtomicBool,
-    geneva_client: GenevaClient,
+    geneva_client: Arc<GenevaClient>,
+    max_concurrent_uploads: usize,
 }
 
 impl GenevaExporter {
     /// Create a new GenavaExporter
     pub fn new(geneva_client: GenevaClient) -> Self {
+        Self::with_concurrency(geneva_client, 10) // Default to 10 concurrent uploads
+    }
+
+    /// Create a new GenavaExporter with custom concurrency level
+    pub fn with_concurrency(geneva_client: GenevaClient, max_concurrent_uploads: usize) -> Self {
```

Comment:
Either use `new_with_concurrency` as the method name or consider using a builder pattern? We could consider using builder pattern if we think we would have more options required later.

```suggestion
    pub fn new_with_concurrency(geneva_client: GenevaClient, max_concurrent_uploads: usize) -> Self {
```